lib/pbench/agent/tool_meister_stop.py

```diff
@@ -65,6 +65,10 @@ def __init__(self, spec: str, benchmark_run_dir: Path, def_host_name: str):
             )
         except FileNotFoundError:
             pass
+        else:
+            # Since this Redis server is locally managed, communication to it
+            # will always be through the "localhost" interface.
+            self.host = "localhost"
```

Comment:
Is this enough? What bothers me is the asymmetry between `tool_meister_start` and `tool_meister_stop`. `start` seems to worry about a bind host name that is *not* `localhost` but is mapped to 127.0.0.1 (see  the comment in l.649 of `tool_meister_start`- not sure what the config file will look like in that case). But the above assumes that `localhost` will *always* be the proper name if redis is locally managed. Why the difference? Why not force the bind host name to be `localhost` when starting? Am I right to worry or am I confused?

Reply:
For "bind" address we generally want to make sure we have something that's externally reachable; but for some reason we *also* tell Redis to listen on `localhost`. The code at line 649 is avoiding a failure if `localhost` is already specified (in some alias), which Redis doesn't like. (That is, we don't *add* `localhost` if it's already there.)

In this case, where we know that we want to connect to the local Redis, `localhost` is a convenience.

Reply:
> For "bind" address we generally want to make sure we have something that's externally reachable; but for some reason we _also_ tell Redis to listen on `localhost`.

It is not "for some reason," it is for _this_ reason: when `pbench-tool-meister-start` is orchestrating the stand-up of the Tool Meister sub-system, we use `localhost` to talk to the Redis server so that the disconnected `pbench-tool-meister-stop` does not have to guess at where the Redis server is listening.  And we don't have the Redis server listen on `0.0.0.0` to avoid any potential accidental conflicts.

> The code at line 649 is avoiding a failure if `localhost` is already specified (in some alias), which Redis doesn't like. (That is, we don't _add_ `localhost` if it's already there.)

This points out a change that should be made to `pbench-tool-meister-start` in another PR, namely, that the addition of `localhost` should only be performed if it orchestrated the sub-system creation.

> In this case, where we know that we want to connect to the local Redis, `localhost` is a convenience.

It is more than a convenience because communicating from `pbench-tool-meister-start` to `pbench-tool-meister-stop` the IP address that should be used would involve a lot more work (don't just parse redis.conf because we might not be able to reach it from the other IP address) and locally managed instances using `localhost` is very natural and efficient.